rules/aip0123/resource_singular.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package aip0123
+
+import (
+	"fmt"
+
+	"github.com/googleapis/api-linter/lint"
+	"github.com/googleapis/api-linter/locations"
+	"github.com/googleapis/api-linter/rules/internal/utils"
+	"github.com/jhump/protoreflect/desc"
+	"github.com/stoewer/go-strcase"
+)
+
+var resourceSingular = &lint.MessageRule{
+	Name: lint.NewRuleName(123, "resource-singular"),
+	OnlyIf: func(m *desc.MessageDescriptor) bool {
+		r := utils.GetResource(m)
+		return r != nil || r.GetType() == ""
+	},
+	LintMessage: func(m *desc.MessageDescriptor) []lint.Problem {
+		r := utils.GetResource(m)
+		l := locations.MessageResource(m)
+		s := r.GetSingular()
+		if s == "" {
+			return []lint.Problem{{
```

Comment:
Can you include a `Suggestion: fmt.Sprintf("singular: %q", strcase.LowerCamelCase(typeName)),`? Will that land in the right spot given the location?

I'd say the same for `plural` but I think it's a tad harder to guess the correct pluralization even with a helper lib :)

Reply:
I'll take a stab. It's a little bit more code but not terrible.